### PR TITLE
implement okta

### PR DESCRIPTION
**Pull request by andrewroth**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class User < ApplicationRecord
-  validates :username, presence: true, uniqueness: true
-  before_validation :pull_attributes, on: :create
+  # validates :username, presence: true, uniqueness: true
+  # before_validation :pull_attributes, on: :create
```
Comment (knutsenm) [inline app/models/user.rb:3]:
```suggestion
  validates :username, presence: true, uniqueness: true
```

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@ def new
   end
 
   def create
-    @user = User.find_or_create_from_auth_hash(auth_hash)
+    @user = User.find_from_auth_hash(auth_hash)
```
Comment (knutsenm) [inline app/controllers/sessions_controller.rb:10]: Was misnamed

File: config/routes.rb [hunk 3]
```diff
@@ -15,7 +15,7 @@
   end
 
   get "/logout", to: "sessions#destroy", as: :destroy_admin_user_session
-  resource :session, only: %i[new create destroy]
+  resource :sessions, only: %i[new create destroy]
```
Comment (knutsenm) [inline config/routes.rb:18]: Resources should be plural

Comment (andrewroth) [reply config/routes.rb:18]: Explained here https://github.com/CruGlobal/ararat/pull/42#discussion_r470376478 why I'd like to keep it singular

Comment (knutsenm) [reply config/routes.rb:18]: In that case, should line 17 be singular as well?

Comment (knutsenm) [reply config/routes.rb:18]: Nope, guide says "singular resources map to plural controllers"

**Pull request merged**